src/oc_erchef/schema/revert/keys_update_trigger.sql

```diff
@@ -1,14 +1,45 @@
 -- Revert keys_update_trigger
-
+--
+-- This keeps the keys table consistent when the clients and user tables update their keys
+--
 BEGIN;
 
+CREATE OR REPLACE FUNCTION add_key() RETURNS TRIGGER AS $add_key$
+   BEGIN
+     IF NEW.public_key IS NOT NULL THEN
+       INSERT INTO keys
+         (id, key_name, public_key, key_version, created_at,  expires_at) VALUES
+         (NEW.id, 'default', NEW.public_key, NEW.pubkey_version, now(), 'infinity'::timestamp);
+     END IF;
+     RETURN NULL; -- result is ignored since this is an AFTER trigger
+   END;
+$add_key$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION update_key() RETURNS TRIGGER AS $update_key$
+   BEGIN
+     IF NEW.public_key IS NOT NULL THEN
+       UPDATE keys SET public_key = NEW.public_key, key_version = NEW.pubkey_version, expires_at = 'infinity'::timestamp
+         WHERE id = NEW.id AND key_name = 'default';
+       INSERT INTO  keys (id, key_name, public_key, key_version, created_at, expires_at)
+         SELECT NEW.id, 'default', NEW.public_key, NEW.pubkey_version, now(), 'infinity'::timestamp
+         WHERE NOT EXISTS (SELECT 1 FROM keys WHERE id = NEW.id AND key_name = 'default');
```

Comment:
Looks like you're trying to do an upsert? Can you use the `ON CONFLICT` clause instead (see https://www.postgresql.org/docs/current/static/sql-insert.html)... not sure what version of PG y'all are on these days).

I saw this was being done above, as well.

Reply:
added a comment can be changes once we drop support for postgres 9.2/9.3

Reply:
Actually maybe I will leave this one as is - since it is the revert of the update trigger.